Sources/FoundationEssentials/Data/Data+Reading.swift

```diff
@@ -25,24 +25,34 @@ import CRT
 import WinSDK
 #endif
 
+func _fgetxattr(_ fd: Int32, _ name: UnsafePointer<CChar>!, _ value: UnsafeMutableRawPointer!, _ size: Int, _ position: UInt32, _ options: Int32) -> Int {
+#if canImport(Darwin)
+    return fgetxattr(fd, name, value, size, position, options)
+#elseif canImport(Glibc)
+    return fgetxattr(fd, name, value, size)
+#else
+    return -1
+#endif
+}
+
 private func readExtendedAttributesFromFileDescriptor(_ fd: Int32, attrsToRead: [String]) -> [String : Data] {
-#if canImport(Darwin) && !NO_FILESYSTEM
+#if !NO_FILESYSTEM
     var output: [String : Data] = [:]
     for key in attrsToRead {
         key.withCString { keyStr in
             let maxXAttrLength = 1000
             withUnsafeTemporaryAllocation(of: CUnsignedChar.self, capacity: maxXAttrLength) { buf in
-                let result = fgetxattr(fd, keyStr, buf.baseAddress, maxXAttrLength, 0, 0)
+                let result = _fgetxattr(fd, keyStr, buf.baseAddress, maxXAttrLength, 0, 0)
```

Comment:
@compnerd - could use your advice here on the Windows version of this code.

Reply:
Windows doesn't really have "extended attributes". The permission model on Windows is pretty rich and allows for per-identity ACLing of different forms. I believe that we currently do no really handle that per se.

Reply:
Ok - the purpose of this xattr is to store some metadata about the encoding of the file. We can't guarantee its presence on non-Darwin platforms, either, so we could just modify this to do nothing on Windows if you think that's correct.

Reply:
Actually, I already did that above in the stub. It looks like we're good to go for now.